crates/bevy_reflect/src/type_path.rs

```diff
@@ -161,3 +167,58 @@ impl<T: TypePath> DynamicTypePath for T {
         Self::module_path()
     }
 }
+
+/// Provides dynamic access to all methods on [`TypePath`].
+#[derive(Clone, Copy)]
+pub struct TypePathVtable {
```

Comment:
Nit: I wonder if there is a friendlier name than `Vtable`. Maybe `Fns` to mimic `ReflectComponentFns`? Or `Lookup`?

`Vtable` is fine haha, but could be a little intimidating.

Reply:
Yeah "vtable" is a language implementation detail imo, not a name for a high level type.

Reply:
This isn't about the Fns imo. This is about the data it provides (static type path impl strings). The fact that we're holding fns at all seems like an implementation detail to me. This is "an owned type that provides TypePath data for TypeInfo". It is effectively a DynamicTypePath impl in its own right (I'm assuming we aren't actually impling that to preserve the static lifetimes).

Things like TypePathProvider, TypePathData, TypeInfoTypePath, etc seem like the right direction.